TennisKata/TennisMatch.cs

```diff
@@ -0,0 +1,123 @@
+﻿namespace TennisKata
+{
+    public class TennisMatch
+    {
+        private readonly Player playerOne;
+        private readonly Player playerTwo;
+        private readonly int setsToWin;
+
+        public TennisMatch(string playerOne, string playerTwo, int setsToWinTheMatch)
+        {
+            this.playerOne = new Player(playerOne);
+            this.playerTwo = new Player(playerTwo);
+            this.setsToWin = setsToWinTheMatch;
+        }
+
+        private bool MatchFinished { get; set; }
+
+        private bool PlayerScored(Player scorer, Player opponent)
```

Comment:
Za długa metoda. Podziel na mniejsze o dobrych nazwach. Metodę powinno się czytac jak książkę. Zacznij od tych bloków z if-ami. Linia 27+ może być opakowana w metodę `HandleTieBreak...`, kolejna to `HandleWonSet`. Te nazwy nie są najlepsze, ale pokazują ideę.